lib/ansible/module_utils/k8s/raw.py

```diff
@@ -101,8 +103,8 @@ def perform_action(self, resource, definition):
 
         self.remove_aliases()
 
-        if definition['kind'].endswith('list'):
```

Comment:
@fabianvf - this would never ever be true, so this is the bit that still needs a fix if the `facts` parameter is removed from `perform_action`

Reply:
Good point. Wonder if we should just make the *List types fail out until we've properly implemented them.

Reply:
Regardless I think we can address that in a separate PR, I think for this PR we can add the `k8s_facts` module and leave the behavior of the other modules unchanged.

Reply:
All changes to raw.py have now been reverted.